### ISSUE TITLE
Signals not handled properly

### ISSUE THREAD

**Issue by robbertkl** [jwilder/dockerize#12]
Running supervisord within dockerize I've noticed the following issues:

When dockerize receives `SIGINT` (e.g. ctrl-c in an interactive docker container) the following happens:

* dockerize outputs `Received signal: interrupt`
* dockerize forwards this signal to its child process (supervisord in my case)
* supervisord gracefully terminates its "programs" and exits
* dockerize outputs `Command finished successfully.`
* (_so far so good, now for the unexpected_)
* dockerize doesn't exit
* after 10 seconds, dockerize outputs `Killing command due to timeout.` and exits

Seems to me the 10 second wait has no function, the child process is already gone (which dockerize noticed) and dockerize should exit right away.

Another issue occurs when dockerize receives `SIGTERM` (e.g. on docker stop):

* dockerize exits immediately without any output
* signal is not forwarded to child process (supervisord) and not waited on
* supervisord (and its running "programs") don't get a chance to gracefully shutdown
* actually the child process (supervisord and its children) is not killed at all and keeps running; however, the container is destroyed along with any processes in it, similar to a `SIGKILL` to those processes (unclean shutdown)

In contrast, when supervisord receives `SIGTERM`, it forwards this to its running programs and waits for them to exit cleanly, killing them with `SIGKILL` if it takes too long. IMO dockerize should do something similar.

**Comment by jwilder** [comment]
@robbertkl The timeout should be fixed in `v0.0.4`.  The `SIGTERM`/`SIGKILL` handling should also be fixed, but you need to make sure your Dockerfile CMC is using the exec form;

```
CMD ["dockerize", "-template", ...]
```

vs.

```
CMD dockerize -template ....
```

The latter actually invokes `/bin/sh dockerize` which will not forward signals to `dockerize` correctly.  You can run `docker exec <container> ps -ef` to see if `dockerize` is actually pid 1 or if `/bin/sh` is.

**Comment by robbertkl** [comment]
Yes, I'm aware of the exec form (and the consequences). Awesome job on the improvements / fixes, I'll give them a spin soon!
